doc/design.md

```diff
@@ -273,6 +295,121 @@ Returns:
     * tbd
 
 
+### **`POST`** `/oeci_login` ?
+
+## Question
+
+What's the workflow for accessing OECI?
+
+Is it a webpage redirect, like the frontend prototype?
+
+Or is it an authentication serverside, like the current Crawler implementation?
+These behaviors need to be reconciled.
+
+I see three design versions:
+
+- Rely on a frontend page redirect, which obtains login access from the OECI directly to allow subsequent requests by Crawler. This requires a Crawler redesign, which currently passes creds to the remote site. It also requires caching a login token provided by OECI. Does the OECI login allow that, e.g. use a cookie?
+
+- Roll the login  into the `/search/` endpoint.
+If so, we need to add oeci credentials to the **`POST`** `/search/ ` body. And also cache the login credentials clientside so we can re-send them on every search (bad).
+
+- Send oeci login credentials to the backend once, and the /search/ endpoint uses them for every /search/ call. That means caching the OECI login creds in the backend.
+
+
+
+### **`POST`** `/api/forgot_password`
+
+Requests a password reset link be sent to a valid (recognized) email address
+
+`POST` body:
+
+- format: `JSON`
+- fields:
+    * email
+
+Status codes:
+
+- `200 OK`
+- `422 UPROCESSABLE ENTITY` email not recognized
+- `5xx` of some kind if the email-send failed
+
+
+### **`POST`** `/api/update_password`
+
+Follows a password reset link (from a sent email), that provides login authorization
+
+## Question: How does this work?
+
+A password reset link provided in an email needs to be a GET request, so the endpoint requires a different method for yielding auth access. If this method is a POST and auth-protected, we need to first obtain an auth_token via the **`GET`** `/api/auth_token` method below.
+
+### Or,
+
+Is this a GET that has been passed an `auth_token` string as a url query param? That's an irregular authorization that is not processed in a POST header, but we're still using it to authorize a password-change. that's weird. I like option 1 better.
+
+
+Headers:
+- `Authorization: <JWT_AUTH_TOKEN>`
+
+`POST` body:
+
+- format: `JSON`
+- fields:
+    * email
+
+Status codes:
+
+- `200 OK`
+- `422 UPROCESSABLE ENTITY`
+
+
+### **`GET`** `/api/auth_token?auth=<special_auth_token>`
+
+Obtains an auth_token using a query param instead of login credentials.
+This is necessary for one option of the Password Reset workflow described above.
+
+Status codes:
+
+- `200 OK`
+- `422 UPROCESSABLE ENTITY` if the auth token query param is invalid
+
+Returns: auth token usable as a regular POST Authorization header.
+
+- format: `JSON`
+- fields:
+    * auth_token
+    
+### **`POST`** `/api/change_user_settings`
+
+Alter user account settings, like password, email address, and admin access, or account information like Name or Group
```

Comment:
Following REST, this is typically done with a PUT to /api/user

Reply:
Cool. I'm well past due to go and learn RESTful design.

Reply:
IMO there actually isn't a ton to know about REST. Main principle that applies here is to use PUT for idempotent state changes and POST for non-idempotent state changes, both excepting cases where DELETE makes more sense. 

I am following the Rails convention which was designed based on REST. In Rails, you can just say `resource :user` in the `routes.rb` file and it will automatically generate RESTful routes for you for the "user" resource.